imod/msw/meteo_mapping.py

```diff
@@ -1,18 +1,43 @@
 from copy import deepcopy
-from typing import Any, Optional, TextIO
+from pathlib import Path
+from typing import Any, Optional, TextIO, cast
 
 import numpy as np
 import pandas as pd
 import xarray as xr
 
+import imod
 from imod.mf6.utilities.regrid import RegridderWeightsCache
 from imod.msw.fixed_format import VariableMetaData
 from imod.msw.pkgbase import MetaSwapPackage
+from imod.msw.utilities.common import find_in_file_list
 from imod.prepare import common
-from imod.typing import GridDataArray, IntArray
+from imod.typing import GridDataArray, Imod5DataDict, IntArray
 from imod.util.regrid_method_type import RegridMethodType
 
 
+def open_first_meteo_grid(mete_grid_path: str | Path, column_nr: int) -> xr.DataArray:
+    """
+    Find first meteo grid path in mete_grid.inp. Only read the first grid, so it
+    can be used to generate meteomappings.
+    """
+    if column_nr not in [2, 3]:
+        raise ValueError("Column nr should be 2 or 3")
+
+    mete_grid_path = Path(mete_grid_path)
+
+    f = open(mete_grid_path, "r")
+    lines = f.readlines()
```

Comment:
Instead of readlines you can use readline to get the first line

Furthermore i wiould suggest using a context to make sure the file is closed when you're done reading it

```
with open(mete_grid_path, 'r') as file:
  first_line = file.readline()
  meteo_filepath = Path(first_line.split(",")[column_nr].replace('"', ""))
```

Reply:
Oomph good one! Missed the context when opening the file 😱